ls1intum/Artemis
# Enhancement/updated c template

### Checklist

### Description


The GBS course updated their C template. I integrated their update into our own template. I also moved the basic compile test from all tests to the structural tests, so that we can have proper sequential test runs if so desired.

### Steps for Testing


1. Log in to Artemis
2. Navigate to Course Administration
3. Create a C programming exercise with sequential runs
4. The template should only have one failing test, but the solution all 11 (successful) (Edit: should be 20 successful tests)
